commit b49f72e7f96d4ed147447428f2ae5b4cea598ca7
Author: Mark Pearson <markpearson@lenovo.com>
Date:   Thu Mar 17 17:40:08 2022 -0400

    platform/x86: think-lmi: Certificate authentication support
    
    Implementation of certificate authentication feature for Lenovo
    platforms. This allows for signed updates of BIOS settings.
    
    Functionality supported:
     - Cert support available check. At initialisation check if BIOS
       supports certification authentication and if a certificate is
       installed. Enable the sysfs nodes appropriately
     - certificate and signature authentication attributes to enable
       a user to install, update and delete a certificate using signed
       signatures
     - certificate_thumbprint to confirm installed certificate details
     - support to go from certificate to password based authentication
     - signature and save_signature attributes needed for setting BIOS
       attributes using certificate authentication.
    
    Tested on X1 Carbon G10 and X1 Yoga G7. This feature is not
    generally available yet but will be released later this year.
    
    Note, I also cleaned up the formating of the GUIDs when I was adding
    the new defines. Hope that's OK to combine in this commit.
    
    Signed-off-by: Mark Pearson <markpearson@lenovo.com>
    Link: https://lore.kernel.org/r/20220317214008.3459-2-markpearson@lenovo.com
    Reviewed-by: Hans de Goede <hdegoede@redhat.com>
    Signed-off-by: Hans de Goede <hdegoede@redhat.com>

diff --git a/drivers/platform/x86/think-lmi.h b/drivers/platform/x86/think-lmi.h
index e46c7f383353..4f69df6eed07 100644
--- a/drivers/platform/x86/think-lmi.h
+++ b/drivers/platform/x86/think-lmi.h
@@ -62,6 +62,10 @@ struct tlmi_pwd_setting {
 	char kbdlang[TLMI_LANG_MAXLEN];
 	int index; /*Used for HDD and NVME auth */
 	enum level_option level;
+	bool cert_installed;
+	char *certificate;
+	char *signature;
+	char *save_signature;
 };
 
 /* Attribute setting details */
@@ -82,6 +86,7 @@ struct think_lmi {
 	bool pending_changes;
 	bool can_debug_cmd;
 	bool opcode_support;
+	bool certificate_support;
 
 	struct tlmi_attr_setting *setting[TLMI_SETTINGS_COUNT];
 	struct device *class_dev;